plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunnerUtils.kt

```diff
@@ -65,6 +69,14 @@ fun runMavenCopyCommands(sourceFolder: File, buildlogBuilder: StringBuilder, log
         // Create shared parameters
         val transformMvnRunner = TransformMavenRunner(project)
         val mvnSettings = MavenRunner.getInstance(project).settings.clone() // clone required to avoid editing user settings
+        val sourceVirtualFile = LocalFileSystem.getInstance().findFileByIoFile(sourceFolder)
+        val module = sourceVirtualFile?.let { ModuleUtilCore.findModuleForFile(it, project) }
+        val moduleSdk = module?.let { ModuleRootManager.getInstance(it).sdk }
+        val sdk = moduleSdk ?: ProjectRootManager.getInstance(project).projectSdk
+
+        if (sdk == null) return MavenCopyCommandsResult.NoJdk // both module SDK and project SDK are null
+
+        mvnSettings.setJreName(sdk.name) // use the SDK we found above
```

Comment:
if they have an SDK defined, shouldn't we respect it?

Reply:
Isn't that what is happening here? We first get the module SDK, and if that is not defined we use the project SDK.

Reply:
This aligns with what we have written in our docs regarding how we expect users to set up their project when using /transform.

Reply:
if they have configured maven to use a specific java version, then i dont think it makes sense to override that

Reply:
If you check line 71 above, we clone the settings to avoid overriding the user's settings. Line 79 is needed because let's say you have project SDK set to JDK17, but module SDK set to JDK8. Without line 79, MavenRunner uses the project SDK and the local build fails. With line 79, MavenRunner uses the module SDK and the local build succeeds.

Reply:
Update: no longer overriding the mvn settings. Just checking when all JDKs (project + module + Maven Runner JDKs) are null, then fail and tell the user.